src/EnergyPlus/DaylightingManager.cc

```diff
@@ -679,120 +658,114 @@ void CalcDayltgCoefficients(EnergyPlusData &state)
         state.dataDaylightingManager->CreateDFSReportFile = false;
     }
 
-    for (ZoneNum = 1; ZoneNum <= state.dataGlobal->NumOfZones; ++ZoneNum) {
-        if (state.dataDaylightingData->ZoneDaylight(ZoneNum).NumOfDayltgExtWins == 0) continue;
+    // TODO MJW: For now preserve order by zone
+    for (int zoneNum = 1; zoneNum <= state.dataGlobal->NumOfZones; ++zoneNum) {
+        for (int controlNum = 1; controlNum <= state.dataDaylightingData->totDaylightingControls; ++controlNum) {
+            auto &thisDaylightControl = state.dataDaylightingData->daylightControl(controlNum);
+            if (thisDaylightControl.zoneIndex != zoneNum) continue;
```

Comment:
Same comment as above. For one thing, can't you just iterate over daylighting controllers, why is the outer `zoneNum` loop necessary?

Reply:
This is temporary.